### DISCUSSION TITLE
With Hive JDBC Drive，I connect to Kyuubi by service discovery mode and return the hive address？

### DISCUSSION THREAD

**Discussion by winfys** [apache/kyuubi#1465] [Q&A] [answered]
```
Class.forName("org.apache.kyuubi.jdbc.KyuubiDriver");
            Connection conn = DriverManager.getConnection("jdbc:hive2://xxx.xxx.xxx.xxx:2181/;serviceDiscoveryMode=zooKeeper;zooKeeperNameSpace=kyuubi","test","test");

```
I use the above address to link to Kyuubi, but it returns the address of Hiveserver, what should I do

**Comment by pan3793** [comment]
What's your expected behavior?

**Reply by winfys** [reply]
The expected return address of Kyuubi is port 10009, but the returned port is 10000

**Reply by pan3793** [reply]
Could you please check the znodes under `/kyuubi`?

**Reply by winfys** [reply]
`[zk: localhost:2181(CONNECTED) 0] ls /kyuubi
[serviceUri=xxx:10009;version=1.3.1-incubating;sequence=0000000025]
[zk: localhost:2181(CONNECTED) 1] get /kyuubi
xxx.xxx.xxx.174
`
This is the information obtained from zookeeper

**Reply by pan3793** [reply]
We have [UT](https://github.com/apache/incubator-kyuubi/blob/master/kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiJdbcDriverHASuite.scala) to ensure that Kyuubi Hive JDBC driver works as expected in HA mode, please provides a reproducible steps if you think it doesn't work.

**Comment by cfmcgrady** [comment, marked as answer]
`zooKeeperNamespace` is case-sensitive. please try

```java
Class.forName("org.apache.kyuubi.jdbc.KyuubiDriver");
Connection conn = DriverManager.getConnection("jdbc:hive2://xxx.xxx.xxx.xxx:2181/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=kyuubi","test","test");

```

**Reply by pan3793** [reply]
> zooKeeperNamespace is case-sensitive.

As we copied the Hive JDBC driver source code to Kyuubi repo, seems we can enhance it to make it be case-insensitive, WDYT @yaooqinn?

**Reply by cfmcgrady** [reply]
There is a bug in [the current official documents](https://kyuubi.apache.org/docs/stable/deployment/high_availability_guide.html#load-balance-mode) and has already been fixed by [PR-1320](https://github.com/apache/incubator-kyuubi/pull/1320).

**Reply by yaooqinn** [reply]
> > zooKeeperNamespace is case-sensitive.
> 
> As we copied the Hive JDBC driver source code to Kyuubi repo, seems we can enhance it to make it be case-insensitive, WDYT @yaooqinn?

I'm ok with this suggestion

**Reply by winfys** [reply]
The problem has been resolved because the hostname of my zookeeper node is incorrectly configured，Thanks
